src/helm/clients/mistral_client.py

```diff
@@ -86,12 +87,35 @@ def make_request(self, request: Request) -> RequestResult:
         """Make a request"""
         completions: List[GeneratedOutput] = []
 
+        prompt: Union[str, List[Dict[str, str]]] = request.prompt
+        if request.multimodal_prompt:
+            # Following https://docs.mistral.ai/capabilities/vision
+            multimodal_content: List[Dict[str, str]] = []
+            for media_object in request.multimodal_prompt.media_objects:
+                if media_object.is_type(IMAGE_TYPE) and media_object.location:
+                    assert media_object.location
+                    if media_object.is_local_file:
+                        from helm.common.images_utils import encode_base64
+
+                        base64_image: str = encode_base64(media_object.location)
+                        image_url = f"data:image/jpeg;base64,{base64_image}"
+                    else:
+                        image_url = media_object.location
+                    multimodal_content.append({"type": "image_url", "image_url": image_url})
+                elif media_object.is_type(TEXT_TYPE):
+                    assert media_object.text
+                    multimodal_content.append({"type": "text", "text": media_object.text})
+                else:
+                    raise ValueError(f"Unrecognized MediaObject type {media_object.type}")
+
+            prompt = multimodal_content
+
         # `num_completions` is not supported, so instead make `num_completions` separate requests.
         for completion_index in range(request.num_completions):
             try:
                 raw_request: MistralAIRequest = {
                     "model": self.mistral_model or request.model_engine,
-                    "prompt": request.prompt,
+                    "prompt": prompt,
```

Comment:
@yifanmai we use `raw_request` as the cache key. Just want to double check with you that `prompt` is okay to use here?

Reply:
I think this is the right thing to do. This assumes that all remote URLs represent immutable files, but I believe this is a correct assumption.